File: golang/src/detectors/improper-certificate-validation/ImproperCertificateValidation.go
package main

import (
	"crypto/tls"
	"net/http"
)

// {fact rule=improper-certificate-validation@v1.0 defects=1}
func improperCertificateValidationNoncompliant(authReq *http.Request) *http.Response {
	// Noncompliant: `InsecureSkipVerify` parameter set to true
	tr := &http.Transport{
		TLSClientConfig: &tls.Config{
			InsecureSkipVerify: true,
			MinVersion:         tls.VersionTLS13,
		},
	}
	client := &http.Client{Transport: tr}
	res, _ := client.Do(authReq)
	return res
}
// {/fact}

// {fact rule=improper-certificate-validation@v1.0 defects=0}
func improperCertificateValidationCompliant(authReq *http.Request) *http.Response {
	// Compliant: `InsecureSkipVerify` parameter set to false
	tr := &http.Transport{
		TLSClientConfig: &tls.Config{
			InsecureSkipVerify: false,
			MinVersion:         tls.VersionTLS13,
		},
	}
	client := &http.Client{Transport: tr}
	res, _ := client.Do(authReq)
	return res
}
// {/fact}

File: ruby/src/detectors/improper-certificate-validation/improper-certificate-validation.rb
=begin
Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0
=end

# {fact rule=improper-certificate-validation@v1.0 defects=1}
require "httparty"

def certificate_validation_noncompliant

  # Noncompliant: SSL certificate validation is disabled.
  HTTParty.get("http://example.com/", verify: false)

end
# {/fact}

# {fact rule=improper-certificate-validation@v1.0 defects=0}
require "httparty"

def certificate_validation_compliant

  # Compliant: SSL certificate validation is enabled.
  HTTParty.get("http://example.com/", verify: true)

end
# {/fact}
